common/src/test/java/io/netty/util/internal/logging/InternalLoggerFactoryTest.java

```diff
@@ -13,49 +13,63 @@
  * License for the specific language governing permissions and limitations
  * under the License.
  */
+
 package io.netty.util.internal.logging;
 
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
 
-import static org.easymock.EasyMock.*;
-import static org.junit.Assert.*;
+import java.util.concurrent.BrokenBarrierException;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.easymock.EasyMock.createMock;
+import static org.easymock.EasyMock.createStrictMock;
+import static org.easymock.EasyMock.expect;
+import static org.easymock.EasyMock.replay;
+import static org.easymock.EasyMock.reset;
+import static org.easymock.EasyMock.verify;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNotSame;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
 
 public class InternalLoggerFactoryTest {
     private static final Exception e = new Exception();
-    private InternalLoggerFactory oldLoggerFactory;
+    private InternalLoggerFactory.InternalLoggerFactoryHolder holder;
```

Comment:
Seems strange to create a mock of the class under test, and using internal implementation details of the class to verify its behavior.  Seems like we should care about testing the `InternalLoggerFactory` interface and not worry about internal implementation details.

Can we have the tests use a real `InternalLoggerFactory` and respect the "only set once" restriction enforced by the interface?